.golangci.yml

```diff
@@ -65,6 +65,8 @@ linters-settings:
       # Forbid everything in syscall except the uppercase constants
       - '^syscall\.[^A-Z_]+$(# Using anything except constants from the syscall package is forbidden )?'
       - '^logrus\.Logger$'
+  usetesting:
+    os-setenv: true
```

Comment:
Is there any reason to enable this explicitly, which is enabled by default, and while we don't explicitly set others. 

I'm even more curious considering that we rarely (never till now?) use `Setenv()`, cause it is incompatible with `t.Parallel`, and we have a decently good environment variables "injection".

Reply:
Also related, why not enabling for instance:
```
os-temp-dir: true
```
which is disabled by default, and we have one occurrence.

Wdyt? 🤔

Reply:
> which is enabled by default,

I actually read their example in the repo as that it is `false` by default :facepalm: and didn't read the exampel above saying it is actually true by default.

I mostly was trying to keep the same as before just transition to newer linters.

I would prefer to open a separate issue to update this and maybe enable more stuff than something else. After all I do this mostly to be able to update to go 1.24 without golangci-lint OOMing.

Does that sound okay?

Reply:
Yup! Let's go!